pkg/storegateway/indexheader/reader_pool_test.go

```diff
@@ -138,3 +125,100 @@ func TestReaderPool_ShouldCloseIdleLazyReaders(t *testing.T) {
 	require.Equal(t, float64(2), promtestutil.ToFloat64(metrics.lazyReader.loadCount))
 	require.Equal(t, float64(2), promtestutil.ToFloat64(metrics.lazyReader.unloadCount))
 }
+
+func TestReaderPool_LoadedBlock(t *testing.T) {
```

Comment:
[nit] Match the name of the method under test:

```suggestion
func TestReaderPool_LoadedBlocks(t *testing.T) {
```